sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -703,45 +705,50 @@ object QueryExecutionErrors {
   }
 
   def dataTypeUnsupportedYetError(dataType: DataType): Throwable = {
-    new UnsupportedOperationException(s"$dataType is not supported yet.")
+    new SparkUnsupportedOperationException(errorClass = "DATA_TYPE_UNSUPPORTED_YET",
+      messageParameters = Array(dataType.toString))
   }
 
   def unsupportedOperationForDataTypeError(dataType: DataType): Throwable = {
-    new UnsupportedOperationException(s"DataType: ${dataType.catalogString}")
+    new SparkUnsupportedOperationException(errorClass = "UNSUPPORTED_OPERATION_FOR_DATA_TYPE",
+      messageParameters = Array(dataType.catalogString))
   }
 
   def inputFilterNotFullyConvertibleError(owner: String): Throwable = {
-    new SparkException(s"The input filter of $owner should be fully convertible.")
+    new SparkException(errorClass = "INPUT_FILTER_NOT_FULLY_CONVERTIBLE",
+      messageParameters = Array(owner), null)
   }
 
   def cannotReadFooterForFileError(file: Path, e: IOException): Throwable = {
-    new SparkException(s"Could not read footer for file: $file", e)
+    new SparkException(errorClass = "CANNOT_READ_FOOTER_FOR_FILE",
+      messageParameters = Array(file.toString), e)
   }
 
   def cannotReadFooterForFileError(file: FileStatus, e: RuntimeException): Throwable = {
-    new IOException(s"Could not read footer for file: $file", e)
+    new SparkIOException(errorClass = "CANNOT_READ_FOOTER_FOR_FILE",
+      messageParameters = Array(file.toString), e)
   }
 
   def foundDuplicateFieldInCaseInsensitiveModeError(
       requiredFieldName: String, matchedOrcFields: String): Throwable = {
-    new RuntimeException(
-      s"""
-         |Found duplicate field(s) "$requiredFieldName": $matchedOrcFields
-         |in case-insensitive mode
-       """.stripMargin.replaceAll("\n", " "))
+    new SparkRuntimeException(errorClass = "FOUND_DUPLICATE_FIELD_IN_CASE_INSENSITIVE_MODE",
+      messageParameters = Array(requiredFieldName, matchedOrcFields))
   }
 
   def failedToMergeIncompatibleSchemasError(
       left: StructType, right: StructType, e: Throwable): Throwable = {
-    new SparkException(s"Failed to merge incompatible schemas $left and $right", e)
+    new SparkException(errorClass = "FAILED_TO_MERGE_IN_COMPATIBLE_SCHEMAS",
+      messageParameters = Array(left.toString, right.toString), e)
   }
 
   def ddlUnsupportedTemporarilyError(ddl: String): Throwable = {
-    new UnsupportedOperationException(s"$ddl is not supported temporarily.")
+    new SparkUnsupportedOperationException(errorClass = "DDL_UNSUPPORTED_TEMPORARILY",
+      messageParameters = Array(ddl))
   }
 
   def operatingOnCanonicalizationPlanError(): Throwable = {
-    new IllegalStateException("operating on canonicalization plan")
+    new SparkIllegalStateException(errorClass = "OPERATING_ON_CANONICALIZATION_PLAN",
```

Comment:
This error class is confusing; can we update it to `CANNOT_OPERATE_ON_CANONICALIZATION_PLAN`?